onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -175,10 +225,16 @@ def schema(self, request, **kwargs):
         self.object = self.get_object()
         if isinstance(self.object.content_object, XForm):
             xform = self.object.content_object
-            headers = xform.get_headers() + ['_id']
+            headers = xform.get_headers()
             self.xform_headers = replace_special_characters_with_underscores(
                 headers)
 
+            # cache a clean list of column headers
+            # using the form id_string as the key
+            column_headers = '{}_{}'.format(
+                TABLEAU_COLUMN_HEADERS, xform.id_string)
+            cache.set(column_headers, self.xform_headers)
```

Comment:
Seems like the `replace_special_characters_with_underscores` doesn't remove the extra fields at all too ? Not sure where that's happening....

Reply:
> Why did you chose to cache the column headers ? I think the cleaned headers are easily re-creatable ?

Yeah, I have removed using cache. since we already have the cform object, I have just re-created how we created headers.

Reply:
> Seems like the `replace_special_characters_with_underscores` doesn't remove the extra fields at all too ? Not sure where that's happening....

Yeah, it was being done in the `self.flatten_xform_columns()` function. BUt I have moved this to this function, seems cleaner.

Reply:
Mhh I think having the `flatten_xform_columns` function separate is fine.. I just didn't see where it was being called, might have missed that part the last time I reviewed this.

_Also wasn't expecting us to cache the raw form of the headers and then flatten the columns...._